### PR TITLE
tests: logging: adjust the structure of logging test cases

### PR DESCRIPTION
**Pull request by enjiamai**
Adjust the logging test case code to make them the same as other test
cases which deal with the CONFIG_USERSPACE used. It does not change
the original test logic, but just the file structure changed.

It also corrects the duplicated testsuite name of logging testcases.

Signed-off-by: Enjia Mai <enjia.mai@intel.com>

### CODE REVIEW

File: tests/subsys/logging/log_core_additional/log2.conf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 CONFIG_MAIN_THREAD_PRIORITY=5
 CONFIG_ZTEST=y
-CONFIG_TEST_USERSPACE=y
+#CONFIG_TEST_USERSPACE=y
```
Comment (SebastianBoe) [inline tests/subsys/logging/log_core_additional/log2.conf:3]: Delete this or else someone might think it was accidentally commented out.

**Review by SebastianBoe — APPROVED** [review]

File: tests/subsys/logging/log_core_additional/log2.conf (hunk 1 above)
Comment (enjiamai) [reply tests/subsys/logging/log_core_additional/log2.conf]: Thank you for correcting this!  @SebastianBoe ~

**Review by chen-png — APPROVED** [review]

**Comment by carlescufi** [comment]
@nordic-krch please take a look

**Comment by cfriedt** [comment]
@nordic-krch - look OK?

**Review by nordic-krch — APPROVED** [review]

**Pull request merged**